### DISCUSSION TITLE
plugins' templates directories are not installed

### DISCUSSION THREAD

**Discussion by AlexandreLanglade** [netbox-community/netbox#9767] [Q&A] [answered]
Hi !

I try to install my own plugins inside my NetBox venv. I don't use a new venv for each plugin. 
After the installation, the *templates* directory is absent in the plugins' dir in site-packages.

My setup.py : 
```python
from setuptools import find_packages, setup

setup(
   name='name',
   version='1',
   description='desc',
   install_requires=[],
   packages=find_packages(),
   include_package_data=True,
   zip_safe=False,
)
```
Py : v3.8.0

Anyone experience something similar ?

**Comment by AlexandreLanglade** [comment]
I've found someone with the same question but he answered himself so I don't know if it's the best solution : 
https://github.com/netbox-community/netbox/discussions/7414

**Comment by candlerb** [comment]


**Reply by AlexandreLanglade** [reply]
@candlerb 
Thank you for your answer, I've updated the question while you were responding.
In fact for the old version of my question : all the plugins were looking in the same plugin's template directory because it was the only one installed in editable mode so the templates dir was obviously here.

**Comment by AlexandreLanglade** [comment]
When I execute find_packages at the root of my plugin project, I can observ that it doesn't see the templates dir :

```python
>>> from setuptools import find_packages
>>> a = find_packages()
>>> a
['pluginname', 'pluginname.api', 'pluginname.migrations']
```

**Comment by AlexandreLanglade** [comment]
If I install all the plugins with the editable option *(the 'develop' argument)* `python3 setup.py develop` as it is shown in the plugin dev tutorial, I have no problem.
But it's not very clean to let them editable when they are terminated.

**Comment by cslingerland** [comment, marked as answer]
I had to use the MANIFEST file just like the answer you suggested earlier. Works well for me.
